core/src/pow/types.rs

```diff
@@ -317,8 +317,10 @@ impl ProofOfWork {
 /// nonces a.k.a. edge indices range from 0 to (1 << edge_bits) - 1
 ///
 /// The hash of the `Proof` is the hash of its packed nonces when serializing
-/// them at their exact bit size. The resulting bit sequence is padded to be
-/// byte-aligned.
+/// them (each nonce) at their exact bit size.
+/// The resulting bit sequence (for each nonce) is padded to be byte-aligned.
```

Comment:
This is quite misleading. You don't pad each nonce. You pad the packed sequence of nonces.